aplijobs/birdwoot
# Feature | Bot starts conversation

[![workerB](https://img.shields.io/endpoint?url=https%3A%2F%2Fworkerb.linearb.io%2Fv2%2Fbadge%2Fprivate%2FU2FsdGVkX1Xi4NlxAHRfirvDQmdqJKwyvBsTbNv6jc%2Fcollaboration.svg%3FcacheSeconds%3D60)](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=Tat7qBT)

First Birdwoot PR 🤩 🚀 

## Multimedia

<details><summary><b>Video</b></summary><br>

https://user-images.githubusercontent.com/22230419/219492281-dfeeb06e-9f94-4ce9-86a6-66b080fb6d6e.mov

</details>

## Description

**Birdwoot**, by default, forces the user to send a message first to start a conversation. This functionality is okay for the Web Plugin workflow, but for referral links purposes, it doesn't. For this case, when a user opens a referral link, the full-screen version of **Birdwoot** should open and after clicking the Start conversation' button, the bot must send the first message to start the interaction with the user.

To accomplish this, we made a hacky move decision jeje so that when a referral param is detected in the URL, no matter its content, we send an empty message to the bot to aware that the conversation has started. This could be improved in the feature by analyzing with more time the code and firing this conversation-started event in a less hacky way 😅 

## How to test it?

1. Run **Birdwoot** locally. Take a look at [this guide](https://www.notion.so/apli/Local-setup-29ddOec1ef8d47ca96d9f8c440b3a6ed?pvs=4)
2. Connect **Birdwoot** with **NAN-E**. Take a look at [this guide](https://www.notion.so/apli/Setup-NAN-E-webhook-1d884637f8243e892ceb834b013810a?ps=4)
3. Go to the Birdwoot dashboard (localhost:3000) and create an inbox, at the end of this process you'll obtain a website token used for that inbox.
4. Now, browse to this address. **Make sure you specify the website token obtained previously in the url**

```
http://localhost:3000/widget?cw_conversation=undefined&website_token=[INBOX_WEBSITE_TOKEN]&locale=en&referral=test#/
```

## Checklist:
